docs/languages/csharp/expression-trees-execution.md

```diff
@@ -0,0 +1,194 @@
+It is not compiled and executable code. If you want to execute
+the .NET code that is represented by an expression tree, you must
+convert it into executable IL instructions. 
+## Lambda Expressions to Functions
+Not all Expression Types can be directly converted into executable
+IL. That sounds like a severe restriction, but in practice, the
+types of expressions that you would want to execute directly are
+all convertible to executable IL. (Think about what it would mean
+to directly execute a `ConstantValueExpression`. Would it mean
+anything useful?) Any expression tree that is a `LamdbaExpression`,
+or a type derived from `LambdaExpression` can be converted to IL.
+The expression type `Expression<TDelegate> where TDelegate : delegate`
+is the only concreted example in the .net Core libraries. It's used
+to represent an expression that maps to any delegate type. Because
+this type maps to a delegate type, the .NET framework can examine
+the expression, and generate IL for an appropriate delegate that
+matches the signature of the lambda expression. 
+
+In most cases, theis creates a simple mapping between an expression,
+and its corresponding delegate. For example, an expression tree that
+is represented by `Expression<Func<int>>` would be converted to a delegate
+of the type `Func<int>`. For a lambda expression with any return type
+and argument list, there exists a delegate type that is the target type
+for the executable code represented by that lamdba expression.
+
+The `LamdbaExpression` type contains `Compile` and `CompileToMethod`
+members that you would use to convert an expression tree to executable
+code. The `Compile` method creates a delegate. The `ConmpileToMethod`
+method updates a `MethodBuilder` object with the IL that represents
+the compiled output of the expression tree. Note that `CompileToMethod`
+is only available on the full desktop framework, not on the 
+.NET Core framework.
+
+Optionally, you can also provide a `DebugInfoGenerator` that will
+receive the symbol debugging information for the generated delegate
+object. This enables you to convert the expression tree into a
+delegate object, and have full debugging information about the
+generated delegate.
+
+You would convert an expression into a delegate using the following
+code:
+
+```cs
+Expression<Func<int>> add = () => 1 + 2;
+var func = add.Compile(); // Create Delegate
+var answer = func(); // Invoke Delegate
+Console.WriteLine(answer);
+```
+
+Notice that the delegate type is based on the expression type. You must
+know the return type and the argument list if you want to use the
+delegate object in a strongly typed manner. The `LambdaExpression.Compile()`
+method returns the `Delegate` type. You will have to cast it to the correct
+delegate type to have any compile-time tools check the argument list of
+return type.
+
+## Execution and Lifetimes
+
+You execute the code by invoking the delegate created
```

Comment:
If I'm understanding this correctly, would it be appropriate to say "[..] by calling `LambdaExpression.Compile()`?